src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -513,11 +513,47 @@ private BlittableJsonReaderObject RecreateCountersIfNeeded(DocumentsOperationCon
                         nonPersistentFlags, flags);
                     return;
                 }
+
+                document = RecreateCountersInfoIfNeeded(context, document, id, ref flags);
```

Comment:
can we reuse the `ChangeSnapshotFlag` ?

Reply:
We don't have `ChangeSnapshotFlag` in v4.2. In addition, as I see it - `ChangeSnapshotFlag` method changes not only the document metadata but the actual data as well, which I'm not sure is a good fit (we are doing so later on `Put` method).

Reply:
You are correct. We need to have this in 4.2 as well.
take a look at the `RevertRevisionWithCounters` test, the metadata keys in 4.2 contain the `@counter-snapshot`, while in 5.2 it works as expected.
cc @efratshenhar